internal/server/volume/server.go

```diff
@@ -221,38 +239,52 @@ func (s *Server) GetVolumeDiskNumber(context context.Context, request *internal.
 		return nil, fmt.Errorf("volume id empty")
 	}
 
-	diskNumber, err := s.hostAPI.GetVolumeDiskNumber(volumeId)
+	diskNumber, err := s.hostAPI.GetDiskNumberFromVolumeID(volumeId)
 	if err != nil {
 		klog.Errorf("failed GetVolumeDiskNumber %v", err)
 		return nil, err
 	}
 
-	response := &internal.VolumeDiskNumberResponse{
+	response := &internal.GetDiskNumberFromVolumeIDResponse{
 		DiskNumber: diskNumber,
 	}
 
 	return response, nil
 }
 
 func (s *Server) GetVolumeIDFromMount(context context.Context, request *internal.VolumeIDFromMountRequest, version apiversion.Version) (*internal.VolumeIDFromMountResponse, error) {
-	klog.V(4).Infof("calling GetVolumeFromMount with request %+v", request)
+	getVolumeIDFromTargetPathRequest := &internal.GetVolumeIDFromTargetPathRequest{
+		TargetPath: request.Mount,
+	}
+	getVolumeIDFromTargetPathResponse, err := s.GetVolumeIDFromTargetPath(context, getVolumeIDFromTargetPathRequest, version)
+	if err != nil {
+		return nil, fmt.Errorf("Forward to GetVolumeIDFromTargetPath failed, err=%+v", err)
+	}
+	volumeIDFromMountResponse := &internal.VolumeIDFromMountResponse{
+		VolumeId: getVolumeIDFromTargetPathResponse.VolumeId,
+	}
+	return volumeIDFromMountResponse, nil
+}
+
+func (s *Server) GetVolumeIDFromTargetPath(context context.Context, request *internal.GetVolumeIDFromTargetPathRequest, version apiversion.Version) (*internal.GetVolumeIDFromTargetPathResponse, error) {
+	klog.V(4).Infof("calling GetVolumeIDFromTargetPath with request %+v", request)
 	minimumVersion := apiversion.NewVersionOrPanic("v1beta1")
 	if version.Compare(minimumVersion) < 0 {
-		return nil, fmt.Errorf("GetVolumeFromMount requires CSI-Proxy API version v1beta1 or greater")
+		return nil, fmt.Errorf("GetVolumeIDFromTargetPath requires CSI-Proxy API version v1beta1 or greater")
```

Comment:
is GetVolumeIDFromTargetPath added to v1beta3?

Reply:
this check was added in `GetVolumeIDFromMount`, I've moved it there instead